packages/amplify-util-mock/src/utils/git-ignore.ts

```diff
@@ -1,16 +1,21 @@
 import * as fs from 'fs-extra';
-import * as path from 'path';
-import { getMockDataDirectory } from './mock-data-directory';
+import * as os from 'os';
 
 export function addMockDataToGitIgnore(context) {
   const gitIgnoreFilePath = context.amplify.pathManager.getGitIgnoreFilePath();
   if (fs.existsSync(gitIgnoreFilePath)) {
-    const gitRoot = path.dirname(gitIgnoreFilePath);
-    const mockDataDirectory = path.relative(gitRoot, getMockDataDirectory(context));
-    let gitIgnoreContent = fs.readFileSync(gitIgnoreFilePath).toString();
-    if (gitIgnoreContent.search(RegExp(`^\s*${mockDataDirectory}\w*$`, 'gm')) === -1) {
-      gitIgnoreContent += '\n' + mockDataDirectory;
-      fs.writeFileSync(gitIgnoreFilePath, gitIgnoreContent);
+    const gitIgnoreContent = fs
+      .readFileSync(gitIgnoreFilePath)
+      .toString()
+      .split(os.EOL);
+    const gitIgnoreMap = gitIgnoreContent.reduce((map, val) => {
+      map.set(val, true);
+      return map;
+    }, new Map<string, boolean>());
```

Comment:
Why do you need a Map? Can't you use a Set since you don't really care about the values